eldritch/mythos.py

```diff
@@ -471,13 +643,195 @@ def get_modifier(self, thing, attribute):
+  def __init__(self):
+    super().__init__("Mythos48", "Isle", "Science", {"hex"}, {"slash", "triangle", "star"})
+
+  def create_event(self, state) -> events.Sequence:
+    seq = super().create_event(state)
+    seq.events.extend(ReturnAndIncrease({"Dimensional Shambler", "Hound"}))
+    return seq
+
+
+class Mythos49(Headline):
+  def __init__(self):
+    super().__init__("Mythos49", "Woods", "Society", {"hex"}, {"slash", "triangle", "star"})
+
+  def create_event(self, state) -> events.Sequence:
+    seq = super().create_event(state)
+    seq.events.extend([
+        events.CloseLocation("Administration", 1),
+        events.CloseLocation("Library", 1),
+        events.CloseLocation("Science", 1),
+        events.AddGlobalEffect(self, "mythos", state.turn_number + 1)
+        # Minor rare interaction with shuffling
+    ])
+
+
+class Mythos50(Headline):
+  def __init__(self):
+    super().__init__("Mythos50", "Woods", "Society", {"circle"}, {"square", "diamond"})
+
+  def create_event(self, state) -> events.Sequence:
+    seq = super().create_event(state)
+    seq.events.append(events.ReleaseMonstersToLocation("Southside", 2))
+    return seq
+
+
+class Mythos51(Environment):
+  def __init__(self):
+    super().__init__("Mythos51", "Graveyard", "Isle", {"square", "diamond"}, {"circle"},
+                     env_type="mystic")
+
+  def get_override(self, thing, attribute):
+    if isinstance(thing, items.Spell) and attribute == "can_use":
+      return False
+    return None
+
+
+class Mythos52(Environment):
+  def __init__(self):
+    super().__init__("Mythos52", "Cave", "Roadhouse", {"plus"}, {"crescent"}, env_type="weather")
+
+  def create_event(self, state) -> events.Sequence:
+    seq = super().create_event(state)
+    seq.events.extend([
+        events.ReturnMonsterFromBoard(None, monster)
+        for monster in state.monsters
+        if isinstance(monster.place, places.CityPlace) and monster.name == "Haunter"
+    ])
+    return seq
+
+  def get_modifier(self, thing, attribute, state):
+    if isinstance(getattr(thing, "place", None), places.CityPlace):
+      if attribute == "sneak":
+        return -1
+      if attribute == "will":
+        return 1
+    return 0
+
+  def get_trigger(self, event, state):
+    # TODO: Override Haunter deployment
+    return None
+
+
+class Mythos53(Headline):
+  def __init__(self):
+    super().__init__("Mythos53", "Square", "Unnamable", {"hex"}, {"slash", "triangle", "star"})
+
+  def create_event(self, state) -> events.Sequence:
+    seq = super().create_event(state)
+    seq.events.extend(ReturnAndIncrease({"Land Squid", "Giant Worm"}))
+    return seq
+
+
+# TODO: Mythos 54, may return from Other Worlds
+
+class Mythos55(Environment):
+  def __init__(self):
+    super().__init__("Mythos55", "Isle", "Science", {"plus"}, {"moon"}, "mystic")
+
+  def get_modifier(self, thing, attribute, state):
+    if (isinstance(thing, monsters.Monster)
+       and thing.has_attribute("undead", state, None)
```

Comment:
TODO: if a player use the red sign to ignore the undead attribute of a monster, does this effect no longer apply?